app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
+
+  def update_users_posts_counter
+    author.increment!(:posts_counter)
+    author.save
```

Comment:
[OPTIONAL]

- [ ] I would recommend using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_users_posts_counter` method every time a new post has been saved to database, then make the `update_users_posts_counter` method private.

```ruby
 # app/models/post.rb
  after_save :update_users_posts_counter

  private

  def update_users_posts_counter
    author.increment!(:posts_counter)
  end
```

Reply:
Thanks for the recommendation!